cosmwasm/contracts/ntt-global-accountant/src/contract.rs

```diff
@@ -0,0 +1,962 @@
+    MessageInfo, Order, Response, StdError, StdResult, Uint256,
+};
+use cw2::set_contract_version;
+use cw_storage_plus::{Bound, KeyDeserialize};
+use ntt_messages::{
+    mode::Mode,
+    ntt::NativeTokenTransfer,
+    transceiver::{Transceiver, TransceiverMessage},
+    transceivers::wormhole::{
+        WormholeTransceiver, WormholeTransceiverInfo, WormholeTransceiverRegistration,
+    },
+    trimmed_amount::TrimmedAmount,
+};
+use serde_wormhole::RawMessage;
+use tinyvec::{Array, TinyVec};
+use wormhole_bindings::WormholeQuery;
+use wormhole_io::TypePrefixedPayload;
+use wormhole_sdk::{
+    accountant_modification::ModificationKind,
+    ntt_accountant as ntt_accountant_module, relayer,
+    vaa::{self, Body, Header, Signature},
+    Chain,
+};
+
+use crate::{
+    bail,
+    error::{AnyError, ContractError},
+    msg::{
+        AllAccountsResponse, AllModificationsResponse, AllPendingTransfersResponse,
+        AllTransceiverHubsResponse, AllTransceiverPeersResponse, AllTransfersResponse,
+        BatchTransferStatusResponse, ExecuteMsg, MigrateMsg, MissingObservation,
+        MissingObservationsResponse, Observation, ObservationError, ObservationStatus, QueryMsg,
+        RelayerChainRegistrationResponse, SubmitObservationResponse, TransferDetails,
+        TransferStatus, SUBMITTED_OBSERVATIONS_PREFIX,
+    },
+    state::{
+        Data, PendingTransfer, TransceiverHub, TransceiverPeer, DIGESTS, PENDING_TRANSFERS,
+        RELAYER_CHAIN_REGISTRATIONS, TRANSCEIVER_PEER, TRANSCEIVER_TO_HUB,
+    },
+    structs::DeliveryInstruction,
+};
+
+// version info for migration info
+const CONTRACT_NAME: &str = "crates.io:ntt-global-accountant";
+const CONTRACT_VERSION: &str = env!("CARGO_PKG_VERSION");
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn instantiate(
+    deps: DepsMut<WormholeQuery>,
+    _env: Env,
+    info: MessageInfo,
+    _msg: Empty,
+) -> Result<Response, AnyError> {
+    set_contract_version(deps.storage, CONTRACT_NAME, CONTRACT_VERSION)
+        .context("failed to set contract version")?;
+
+    Ok(Response::new()
+        .add_attribute("action", "instantiate")
+        .add_attribute("owner", info.sender)
+        .add_attribute("version", CONTRACT_VERSION))
+}
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn migrate(_deps: DepsMut<WormholeQuery>, _env: Env, _msg: MigrateMsg) -> StdResult<Response> {
+    Ok(Response::default())
+}
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn execute(
+    deps: DepsMut<WormholeQuery>,
+    _env: Env,
+    info: MessageInfo,
+    msg: ExecuteMsg,
+) -> Result<Response, AnyError> {
+    match msg {
+        ExecuteMsg::SubmitObservations {
+            observations,
+            guardian_set_index,
+            signature,
+        } => submit_observations(deps, info, observations, guardian_set_index, signature),
+
+        ExecuteMsg::SubmitVaas { vaas } => submit_vaas(deps, info, vaas),
+    }
+}
+
```

Comment:
Can remove this in favor of `TRIMMED_DECIMALS` from https://github.com/wormhole-foundation/example-native-token-transfers/blob/main/solana/modules/ntt-messages/src/trimmed_amount.rs#L17.